src/shims/env.rs

```diff
@@ -54,33 +63,109 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn getenv(&mut self, name_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, Scalar<Tag>> {
         let this = self.eval_context_mut();
         let target_os = this.tcx.sess.target.target.target_os.as_str();
```

Comment:
`as_str` calls on `String`? Seems a bit suspicious, I guess this wants to do `&....target_os[..]`?

Reply:
I find a postfix method call much more readable though.

Reply:
Please correct me if I'm wrong :
I've been aware that calling `string.as_str()` returns a `&str` type and is equivalent to using `&string[..]` (both ways borrowing the `string` immutably)
I thought using either way is okay (since they basically do the same thing)

Reply:
Having been through the `x.as_slice()`/`x.as_str()` -> `&x[..]` transition personally, I never understand why people would want their old methods back.

There's nothing semantic here, you're just grabbing a view. That's why we have auto-deref for method calls/operators and deref coercions.

Explicit slicing is very rarely needed, and if you try not to use the methods, most of the time you don't need anything at all.

Case in point, [you don't need an `&str`, `&String` works fine](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4d7dc70e9fe1358cf62f9aff095bd2e4).

Meaning you can use do this:
```rust
let target_os = &this.tcx.sess.target.target.target_os;
```

Which is the normal thing you'd do for *any other* type in Rust, if you wanted a shorthand for it that doesn't move or clone the original.

Reply:
Thank you for the clarification! 😸 I can replace invocations of `.as_str()` to `&` in the next commit.